packages/react/src/components/form/form.tsx

```diff
@@ -0,0 +1,195 @@
+
+const [FormContext, useFormContext] = createContext<FormContext>({
+  name: "FormContext",
+  strict: false,
+})
+
+export { FormContext, useFormContext }
+
+export interface FormRootProps
+  extends HTMLStyledProps<"form">,
+    ThemeProps<FormStyle> {
+  /**
+   * An object managing the disabled state of each field.
+   */
+  disabled?: boolean | { [key: string]: boolean | undefined }
+  /**
+   * An object managing the error message of each field.
+   */
+  errorMessage?: { [key: string]: string | undefined }
+  /**
+   * An object managing the helper message of each field.
+   */
+  helperMessage?: { [key: string]: string | undefined }
+  /**
+   * An object managing the invalid state of each field.
+   */
+  invalid?: boolean | { [key: string]: boolean | undefined }
+  /**
+   * The optional indicator to use for each field.
+   */
+  optionalIndicator?: ReactNode
+  /**
+   * An object managing the read only state of each field.
+   */
+  readOnly?: boolean | { [key: string]: boolean | undefined }
+  /**
+   * An object managing the replace state of each field.
+   */
+  replace?: boolean | { [key: string]: boolean | undefined }
+  /**
+   * An object managing the required state of each field.
+   */
+  required?: boolean | { [key: string]: boolean | undefined }
+  /**
+   * The required indicator to use for each field.
+   */
+  requiredIndicator?: ReactNode
+}
+
+const {
+  PropsContext: FormPropsContext,
+  usePropsContext: useFormPropsContext,
+  withContext,
+  withProvider,
+} = createSlotComponent<FormRootProps, FormStyle, FormContext>(
+  "form",
+  formStyle,
+)
+
+export { FormPropsContext, useFormPropsContext }
+
+/**
+ * `Form` is a component used to group multiple form elements.
+ *
+ * @see https://yamada-ui.com/docs/components/form
+ */
+export const FormRoot = withProvider<"form", FormRootProps>(
+  ({
+    id,
+    children,
+    disabled,
+    errorMessage,
+    helperMessage,
+    invalid,
+    noValidate = true,
+    optionalIndicator,
+    readOnly,
+    replace,
+    required,
+    requiredIndicator,
+    ...rest
+  }) => {
+    const uuid = useId()
+
+    id ??= uuid
+
+    const context = useMemo(
+      () => ({
+        id,
+        disabled,
+        errorMessage,
+        helperMessage,
+        invalid,
+        optionalIndicator,
+        readOnly,
+        replace,
+        required,
+        requiredIndicator,
+      }),
+      [
+        id,
+        disabled,
+        errorMessage,
+        helperMessage,
+        invalid,
+        readOnly,
+        replace,
+        required,
+        optionalIndicator,
+        requiredIndicator,
+      ],
+    )
+
+    return (
+      <FormContext value={context}>
+        <styled.form id={id} noValidate={noValidate} {...rest}>
+          {children}
+        </styled.form>
+      </FormContext>
+    )
+  },
+  "root",
+)()
+
+export interface FormHeaderProps extends HTMLStyledProps {}
+
+export const FormHeader = withContext<"div", FormHeaderProps>("div", "header")()
```

Comment:
Shouldn't we use more semantic tags, like we are doing with the Card component?

```suggestion
export interface FormHeaderProps extends HTMLStyledProps<"header"> {}

export const FormHeader = withContext<"header", FormHeaderProps>("header", "header")()
```